src/BiDirectionalLinkedListImp.h

```diff
@@ -0,0 +1,152 @@
+//
+// Created by shvmo on 02/05/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_BIDIRECTIONALLINKEDLISTIMP_H
+#define DATASTRUCTURESWET1_BIDIRECTIONALLINKEDLISTIMP_H
+
+#include <cassert>
+#include "BiDirectionalLinkedList.h"
```

Comment:
ok remove all cyclic includes from all the generic ADT's `<ADT>Imp.h` files

Reply:
But cyclic includes are okay? That's why we have ifdef.

Reply:
we have ifdef's so when 2 libraries we use share usage of a third library then there will be no conflicts. 🤔 
but if you believe it's more readable then fine